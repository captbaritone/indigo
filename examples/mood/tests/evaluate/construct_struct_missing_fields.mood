struct Foo {
  a: i32,
  b: i32,
  c: i32,
}

pub fn test(): i32 {
    Foo { a: 1_i32 };
    0_i32
}