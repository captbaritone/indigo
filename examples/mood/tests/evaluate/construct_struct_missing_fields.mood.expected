-----------------
INPUT
----------------- 
struct Foo {
  a: i32,
  b: i32,
  c: i32,
}

pub fn test(): i32 {
    Foo { a: 1_i32 };
    0_i32
}
-----------------
OUTPUT
-----------------
Error: Missing struct field(s): "b", "c".:
 --> construct_struct_missing_fields.mood:8:20

  | pub fn test(): i32 {
8 |     Foo { a: 1_i32 };
  |                    ^ "Foo" is missing the field(s) "b", "c".
  |     0_i32
